datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -316,6 +326,71 @@ impl BatchPartitioner {
                             Ok((partition, batch))
                         });
 
+                    Box::new(it)
+                }
+                BatchPartitionerState::HashSelectVector {
+                    random_state,
+                    exprs,
+                    num_partitions,
+                    hash_buffer,
+                } => {
+                    let timer = self.timer.timer();
+                    let arrays = exprs
+                        .iter()
+                        .map(|expr| expr.evaluate(&batch)?.into_array(batch.num_rows()))
+                        .collect::<Result<Vec<_>>>()?;
+
+                    create_hashes(&arrays, random_state, hash_buffer)?;
+
+                    let hash_vector = hash_buffer
+                        .iter()
+                        .map(|hash| *hash % *num_partitions as u64)
+                        .collect::<Vec<_>>();
+                    let mut fields = batch
+                        .schema()
+                        .fields()
+                        .iter()
+                        .map(|f| Arc::clone(f))
+                        .collect::<Vec<_>>();
+                    fields.push(Arc::new(Field::new(
+                        "selection",
+                        DataType::Boolean,
+                        false,
+                    )));
+                    // Finished building index-arrays for output partitions
+                    timer.done();
+
+                    // Borrowing partitioner timer to prevent moving `self` to closure
+                    let partitioner_timer = &self.timer;
+
+                    let it = (0..*num_partitions).into_iter().map(move |partition| {
+                        // Tracking time required for repartitioned batches construction
+                        let _timer = partitioner_timer.timer();
+                        let select_vector = hash_vector
+                            .iter()
+                            .map(|&hash| hash == partition as u64)
```

Comment:
You can `collect` this directly to a `BooleanArray`

Reply:
Nice suggestion! Thanks.